core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -427,76 +434,51 @@ protected boolean checkBlocks() {
             return true;
         }
 
-        protected ShardRouting resolvePrimary(ShardIterator shardIt) {
-            // no shardIt, might be in the case between index gateway recovery and shardIt initialization
-            ShardRouting shard;
-            while ((shard = shardIt.nextOrNull()) != null) {
-                // we only deal with primary shardIt here...
-                if (shard.primary()) {
-                    return shard;
-                }
-            }
-            return null;
+        protected void performReroute(final ShardRouting primary) {
```

Comment:
Added them to make them more readable w.r.t. what they do, I reverted it and added comments instead